macaw-ppc-syntax/src/Data/Macaw/PPC/Symbolic/Syntax.hs

```diff
@@ -0,0 +1,247 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE ImplicitParams #-}
+{-# LANGUAGE ImportQualifiedPost #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeOperators #-}
+
+-- | 'LCSC.ParserHooks' for parsing macaw-ppc-symbolic CFGs.
+module Data.Macaw.PPC.Symbolic.Syntax
+  ( ppcParserHooks
+  , ppc32ParserHooks
+  , ppc64ParserHooks
+    -- * Types
+  , ppcTypeParser
+    -- * Operations
+  , parseRegs
+  , parseReg
+  , ppcAtomParser
+  ) where
+
+import Control.Applicative ( empty )
+import Control.Monad qualified as Monad
+import Control.Monad.IO.Class (MonadIO)
+import Control.Monad.State.Strict (MonadState)
+import Control.Monad.Writer.Strict (MonadWriter)
+import Data.Text qualified as Text
+import GHC.TypeNats ( type (<=) )
+
+import Data.Macaw.Memory qualified as DMM
+import Data.Macaw.Symbolic qualified as DMS
+import Data.Macaw.PPC qualified as PPC
+import Data.Macaw.PPC.Symbolic qualified as PPCS
+import Data.Macaw.PPC.Symbolic.Regs qualified as PPCR
+import Data.Parameterized.Context qualified as Ctx
+import Data.Parameterized.Some (Some(..))
+import Lang.Crucible.CFG.Expr as Expr
+import Lang.Crucible.CFG.Reg (Atom, Stmt)
+import Lang.Crucible.CFG.Reg qualified as Reg
+import Lang.Crucible.Syntax.Atoms qualified as LCSA
+import Lang.Crucible.Syntax.Concrete qualified as LCSC
+import Lang.Crucible.Syntax.Monad qualified as LCSM
+import Lang.Crucible.Types qualified as LCT
+import SemMC.Architecture.PPC qualified as SAP
+import What4.ProgramLoc (Posd(..))
+
+-- | Parser hooks for 32-bit macaw-ppc-symbolic CFGs
```

Comment:
Is this actually 32-bit specific? Looks generic to me!

Reply:
Nope, that is copy-paste detritus. Now fixed.